src/Build/BuildCheck/Infrastructure/TracingReporter.cs

```diff
@@ -6,13 +6,20 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Microsoft.Build.BuildCheck.Utilities;
+using Microsoft.Build.Experimental.BuildCheck;
 
 namespace Microsoft.Build.BuildCheck.Infrastructure;
 
 internal class TracingReporter
 {
     internal Dictionary<string, TimeSpan> TracingStats { get; } = new();
 
+    // Infrastructure time keepers, examples for now
+    internal TimeSpan analyzerAcquisitionTime;
+    internal TimeSpan analyzerSetDataSourceTime;
+    internal TimeSpan newProjectAnalyzersTime;
```

Comment:
Lets turn those into autoproperties

Reply:
Or better - let's expose only an AddXXX methods for those - that way we won't accidentaly overwrite the counter in the calling code

Reply:
Lets make those private, so that that the Add methods cannot be bypassed.